src/code.rs

```diff
@@ -448,23 +456,43 @@ fn build_imports(table: &ParsedTableMacro, config: &GenerationConfig) -> String
         .collect::<Vec<String>>()
         .join("\n");
     #[cfg(feature = "async")]
-    let async_imports = if table_options.get_async() { "\nuse diesel_async::RunQueryDsl;" } else { "" };
+    let async_imports = if table_options.get_async() {
+        "\nuse diesel_async::RunQueryDsl;"
+    } else {
+        ""
+    };
     #[cfg(not(feature = "async"))]
     let async_imports = "";
+    let fns_imports = if config.with_fns {
+        indoc! {"
+        use diesel::QueryResult;
+        "}
+    } else {
+        ""
+    };
+    let connection_type_alias = if config.with_fns {
+        format!(
+            "\ntype Connection = {connection_type};",
+            connection_type = config.connection_type,
+        )
+    } else {
+        "".to_string()
+    };
     format!(
         indoc! {"
         use crate::diesel::*;
         use crate::schema::*;
```

Comment:
`use crate::schema::*;{fns_imports}` so we aren't left with an empty line when they don't use this feature

Reply:
also, let's rename fns_imports

I know it's an abbreviation for functions imports, or something like that, but it's not very clear and generally abbreviations should be avoided in variable names unless they're completely obvious (like gen = generate)

Reply:
Done.